src/OpenSpaceToolkit/Astrodynamics/Trajectory/Propagator.cpp

```diff
@@ -70,17 +70,33 @@ std::ostream& operator<<(std::ostream& anOutputStream, const Propagator& aPropag
 
 bool Propagator::isDefined() const
 {
-    return numericalSolver_.isDefined();
+    return this->numericalSolver_.isDefined() && this->coordinatesBrokerSPtr_ != nullptr &&
+           !this->dynamicsContexts_.isEmpty();
 }
 
-Size Propagator::getNumberOfCoordinates() const
+const Shared<CoordinatesBroker>& Propagator::accessCoordinatesBroker() const
 {
-    return this->coordinatesBrokerSPtr_->getNumberOfCoordinates();
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("Propagator");
+    }
+
+    return this->coordinatesBrokerSPtr_;
 }
 
-const Shared<CoordinatesBroker>& Propagator::accessCoordinatesBroker() const
+const NumericalSolver& Propagator::accessNumericalSolver() const
 {
-    return this->coordinatesBrokerSPtr_;
+    if (!this->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("Propagator");
+    }
+
+    return this->numericalSolver_;
+}
+
+Size Propagator::getNumberOfCoordinates() const
+{
+    return this->coordinatesBrokerSPtr_->getNumberOfCoordinates();
```

Comment:
```suggestion
    return this->accessCoordinatesBroker()->getNumberOfCoordinates();
```

so it uses the `isDefined` check

Reply:
Ah nice yeah. That's a good call.